ApiReview/API.Polly.Core/NoDocs/Polly.Retry/RetryBackoffType.cs

```diff
@@ -0,0 +1,11 @@
+// Assembly 'Polly.Core'
+
+namespace Polly.Retry;
+
+public enum RetryBackoffType
+{
+    Constant = 0,
+    Linear = 1,
+    Exponential = 2,
+    ExponentialWithJitter = 3
+}
```

Comment:
Is it worth adding a way to use jitter with more of the backoff types?

We could either have:

```diff
public enum RetryBackoffType
{
    Constant,
+   ConstantWithJitter,
    Linear,
+   LinearWithJitter,
    Exponential,
    ExponentialWithJitter,
}
```

Or would could remove `ExponentialWithJitter` and instead make having jitter or not its own property.

```diff
public enum RetryBackoffType
{
    Constant,
    Linear,
    Exponential,
-   ExponentialWithJitter,
}

public class RetryStrategyOptions
{
+   public bool JitterEnabled { get; set; }
}
```

Context from https://github.com/App-vNext/Polly/issues/1365#issuecomment-1619492738.

Reply:
I like the latter option. Especially since people would likely start off by declaring their strategy options.

Reply:
Second option + `UseJitter`